contrib/gimp.lua

```diff
@@ -0,0 +1,179 @@
+      the layers used.  Where you save them is up to you.
+
+    BUGS, COMMENTS, SUGGESTIONS
+    * Send to Bill Ferguson, wpferguson@gmail.com
+]]
+
+local dt = require "darktable"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("gimp",dt.configuration.config_dir.."/lua/")
+
+-- Thanks to http://lua-users.org/wiki/SplitJoin for the split and split_path functions
+local function split(str, pat)
+   local t = {}  -- NOTE: use {n = 0} in Lua-5.0
+   local fpat = "(.-)" .. pat
+   local last_end = 1
+   local s, e, cap = str:find(fpat, 1)
+   while s do
+      if s ~= 1 or cap ~= "" then
+        table.insert(t,cap)
+      end
+      last_end = e+1
+      s, e, cap = str:find(fpat, last_end)
+   end
+   if last_end <= #str then
+      cap = str:sub(last_end)
+      table.insert(t, cap)
+   end
+   return t
+end
+
+local function split_path(str)
+   return split(str,'[\\/]+')
+end
+
+local function get_filename(str)
+  parts = split_path(str)
+  return parts[#parts]
+end
+
+local function basename(str)
+  return string.sub(str,1,-4)
+end
+
+local function _(msgid)
+    return gettext.dgettext("gimp", msgid)
+end
+
+local function checkIfBinExists(bin)
+  local handle = io.popen("which "..bin)
+  local result = handle:read()
+  local ret
+  handle:close()
+  if (result) then
+    dt.print_error("true checkIfBinExists: "..bin)
+    ret = true
+  else
+    dt.print_error(bin.." not found")
+    ret = false
+  end
+  return ret
+end
+
+local function show_status(storage, image, format, filename,
+  number, total, high_quality, extra_data)
+    dt.print(string.format(_("Export Image %i/%i"), number, total))
+end
+
+local function gimp_edit(storage, image_table, extra_data) --finalize
+  if not checkIfBinExists("gimp") then
+    dt.print_error(_("gimp not found"))
+    return
+  end
+
+  -- list of exported images 
+  local img_list
+
+   -- reset and create image list
+  img_list = ""
+
+  for _,exp_img in pairs(image_table) do
+    img_list = img_list ..exp_img.. " "
+  end
+
+  dt.print(_("Launching gimp..."))
+
+  local gimpStartCommand
+  gimpStartCommand = "gimp "..img_list
+  
+  dt.print_error(gimpStartCommand)
+
+  coroutine.yield("RUN_COMMAND", gimpStartCommand)
+
+  -- for each of the exported images
+  -- find the matching original image
+  -- then move the exported image into the directory with the original  
+  -- then import the image into the database which will group it with the original
+  -- and then copy over any tags other than darktable tags
+
+  for _,exp_img in pairs(image_table) do
+    local fname = get_filename(exp_img)
+    for _,orig_img in pairs(dt.gui.action_images) do
+      if basename(fname) == basename(orig_img.filename) then
+        os.execute("mv "..exp_img.." "..orig_img.path)
```

Comment:
I've just tested the script and it failed because my orig_img.path has a space in it. I think you should use os.rename here.

http://www.lua.org/manual/5.1/manual.html#pdf-os.rename